### DISCUSSION TITLE
기기별  화면비 적용이 다 따로따로 되나요? 화면비 적용을 모든 기기에 동일하게 하고싶습니다.

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#585] [5. Q&A]
과거 제페토에서는 기기에 상관없이 1980X1080의 화면 비율의 월드를 볼 수가 있어서 해당 비율로 캔버스 자체를 맞추고 월드를 만들었는데,

이번에 아이폰과 아이패드로 테스트해보니 아이폰은 1980X1080에 맞게 텍스쳐가 잘 나오는데, 아이패드Pro 2021버전 에서는 1980X1080이 아닌 전체화면에 맞게 확장되어 나와 현제 제작한 텍스쳐가 늘어나고 찌부라드는 등의 영향이 있습니다.

해당 설정을 제페토측에서 무조건 꽉차는 화면으로만 나오게 한건가요? 수정이 가능할까요? 혹은 개발자 측에서 해당사항을 조율할 수 있을까요?

**Comment by ted002** [comment]
요점은 고정된 16:9의 화면비가 가지고 싶습니다!

**Comment by Joyounwoo** [comment]
캔버스 스칼라 컴포넌트를 조절하시면 됩니다.
<img width="464" alt="스크린샷 2022-04-05 오후 5 38 09" src="https://user-images.githubusercontent.com/49576098/161713754-5f2f78a1-47f9-41b5-892a-972077a1a549.png">
위 이미지처럼 값을 설정하시면 1920x1080 기준으로 설정되며, 다른 화면 비율에서도 이미지가 겹치거나 찌그러지는 일이 발생하시진 않을 겁니다.
덤으로, UI 오브젝트들의 앵커 수치와 피봇 수치를 잘 조절하시면 더 좋습니다.

**Reply by ted002** [reply]
해당부분을 해봤는데, 변화는 없었습니다. 패드에서는 계속 찌그러지더라구요...

**Reply by Joyounwoo** [reply]
찌그러지는 모습과 해당 UI 오브젝트의 인스펙터 창을 확인해봐야 알 수 있습니다.
값을 어떻게 하느냐에 따라 얼마든지 형태가 바뀌게 만들 수 있습니다. 해당 오브젝트 부모의 상태에 따라서도 마찬가지구요.

**Reply by ted002** [reply]
으음...알겠습니다 이게 텍스쳐는 사정이 있어서 올리기 힘들어서...잠시만 기다려주세요

**Comment by ted002** [comment]
![Zepeto - SingleBlock_Test - PC, Mac   Linux Standalone - Unity 2020 3 9f1 Personal _DX11_ 2022-04-06 오후 5_55_47](https://user-images.githubusercontent.com/73985764/161937382-7ae93730-18ac-4d79-8883-079da65f5a60.png)

제가 원하는 정상적인 상태의 텍스쳐(우측 사각형이 정사각형임)

![Zepeto - SingleBlock_Test - PC, Mac   Linux Standalone - Unity 2020 3 9f1 Personal _DX11_ 2022-04-06 오후 5_55_54](https://user-images.githubusercontent.com/73985764/161937450-3fcf58ab-3ba6-4cc0-a833-aa94f487692a.png)

아이패드등에서 보이는 텍스쳐의 찌그러짐(우측 사각형이 세로 직사각형임.)

![Zepeto - SingleBlock_Test - PC, Mac   Linux Standalone - Unity 2020 3 9f1 Personal _DX11_ 2022-04-06 오후 6_02_02](https://user-images.githubusercontent.com/73985764/161938175-12b5022b-d766-44e7-ab19-9d7cbf1b1ef6.png)

갤럭시 A21에서 보이는 텍스쳐의 찌그러짐(우측 사각형이 가로 직사각형임.)

![질문 1](https://user-images.githubusercontent.com/73985764/161938451-99321ed8-372d-4c87-9dc8-e3f27993dac0.png)
![질문 2](https://user-images.githubusercontent.com/73985764/161938469-8676b457-3d9f-44f2-9187-c94c5c82a852.png)

첫번째는 전체적인 UI의 캔버스 사진이고
두번째는 prefeb으로 만든 카메라의 전용UI 캔버스입니다.
찌그러지는 텍스쳐는 카메라 전용 UI캔버스에 적용되어 있습니다

혹시 확인하시는데 필요한 텍스쳐가 더 필요하시면 말씀주세요

**Comment by Joyounwoo** [comment]
이미지의 렉트 트랜스폼을 확인해보셔야 할 것 같습니다.

![image](https://user-images.githubusercontent.com/49576098/162011629-51750dec-5d1a-44cb-96cc-360ecad4115a.png)
앵커 설정이 이 형태가 아닌,

![image](https://user-images.githubusercontent.com/49576098/162011735-1fa255bd-72d9-4146-a65d-f50631a03e5d.png)
이런 형태로, 스트레치로 되어 계셔서 캔버스 크기 변환에 따라 같이 변동되고 있을 것으로 예상됩니다.

앵커 프리셋을 스트레치가 아닌 다른 값으로 수정하시면 되겠습니다.

**Reply by ted002** [reply]
해당 방법을 적용해 보았으나 이번에는 프레임과 화면 사이에 빈 틈이 생깁니다.

가로가 더 긴 화면의 경우에는 가로 양 옆에, 세로가 긴 화면에는 세로 위아래에 빈 틈이 생깁니다.

결국 큰 의미는 없었습니다. 제가 원하는건 정확히 16:9의 종횡비를 가진 프레임에 맞게 스크린샷을 가지는 것인데, 이런식이라면 빈 양 옆은 패널을 활용해 검은색으로 채운다고 해도, 결국 찍고 나오는 텍스쳐는 16:9의 종횡비 텍스쳐가 아닌, 16:9의 종횡비 텍스쳐 + 검은색 블록들입니다.

**Reply by Joyounwoo** [reply]
빈 공간은 앵커 설정을 통해 수정하실 수 있습니다.
구글에 '유니티 렉트트랜스폼 앵커 프리셋' 검색해보시면, 말씀하신 틈에 대해 원하는 형태로 나오게 수정할 수 있는 방법들이 자세히 나와있습니다. 세부적인 설정 부분은 해당 내용 살펴보시면 작업하시면 됩니다.

**Reply by ted002** [reply]
답변 너무너무 감사드립니다. 사실 여러모로 막혀서 계속 수정중에 있는데...아직 해결이 안되었습니다. 그래도 여러가지 해결말씀 감사드립니다.
